File: fsharp/armstrong-numbers/ArmstrongNumbers.fs
module ArmstrongNumbers

let isArmstrongNumber (number: int): bool =
    let digits = 
        number 
        |> Array.unfold (function 
            | 0 -> None 
            | x -> Some (x % 10, x / 10))

    let n = Seq.length digits

    digits 
    |> Array.sumBy (fun x -> pown x n) 
    |> (=) number

File: typescript/armstrong-numbers/armstrong-numbers.ts
export default class {
    static isArmstrongNumber(n: number): boolean {
        const digits = n.toString().split('')
        const armstrongSum = digits
            .reduce((acc, curr) => acc + Number(curr) ** digits.length, 0)

        return armstrongSum === n
    }

    static isArmstrongNumber2(n: number): boolean {
        const digits = this.getDigits(n)
        const armstrongSum = digits
            .reduce((acc, curr) => acc + curr ** digits.length, 0)

        return armstrongSum === n
    }

    static getDigits(n: number): number[] {
        const retVal = []

        while (n > 0) {
            const temp = n % 10
            retVal.push(temp)
            n = Math.floor(n / 10)
        }

        return retVal
    }
}
